apps/prairielearn/python/prairielearn.py

```diff
@@ -67,12 +67,21 @@ class QuestionData(TypedDict):
     extensions: dict[str, Any]
     num_valid_submissions: int
     manual_grading: bool
+    answers_names: dict[str, bool]
 
 
 class ElementTestData(QuestionData):
     test_type: Literal["correct", "incorrect", "invalid"]
 
 
+def check_answers_names(data: QuestionData, name: str) -> None:
+    """Checks that answers names are distinct using property in data dict."""
+    if name in data["answers_names"]:
+        raise KeyError('Duplicate answer-name attribute: "{name}"')
+    else:
+        data["answers_names"][name] = True
```

Comment:
@nwalters512 is it ok to throw an error here, or are we worried it is going to break questions in production?

It seems like it would be nice to have a method of providing warnings during question rendering that would show up on the instructor preview page but not for students.

Reply:
I think it's fine to throw an error, any question that has a duplicate answers name should be considered broken (i.e., things won't grade right, and rendering will be weird).

For your second item, I've already created an issue: https://github.com/PrairieLearn/PrairieLearn/issues/7337

There's a few instances where it would be nice to display warnings for the question that are just reading from standard error (or similar).